contrib/externalMatchfinder/main.c

```diff
@@ -0,0 +1,93 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define ZSTD_STATIC_LINKING_ONLY
+#include "zstd.h"
+#include "zstd_errors.h"
+#include "matchfinder.h" // simpleExternalMatchFinder, simpleExternalMatchStateDestructor
+
+int main(int argc, char *argv[]) {
+    size_t res;
+
+    if (argc != 2) {
+        printf("Usage: exampleMatchfinder <file>\n");
+        return 1;
+    }
+
+    ZSTD_CCtx* zc = ZSTD_createCCtx();
+
+    int simpleExternalMatchState = 0xdeadbeef; // @nocommit
+
+    // Here is the crucial bit of code!
+    res = ZSTD_registerExternalMatchFinder(
```

Comment:
It seems to me to each application has to call this API to enable MatchFinder which will add lots of cost to application to adopt to new Matchfinder API. 
Can we make it more smooth? 
Say if ZSTD is compiled with Matchfinder enabled, the 1st ZSTD_creatCCtx() call will call ZSTD_loadMatchFinder() will will try dynamic load match-finder library and call the symbol ZSTD_registerMatherFinder() defined in that library. 
The match-finder library should be put into specfic ZSTD_MATCHFINDER_DIR. 
Then if the register success, it will call the registered function like ZSTD_Ctx->external_matchfinder_init().

Reply:
@Cyan4973 do you have anything to add here?